Added listener options and dialer options

wanderer opened the pull request.

wanderer committed:
Update base-test.js

diasdavid commented on tests/base-test.js:
```diff
@@ -1,6 +1,6 @@
 var multiaddr = require('multiaddr')
 
-module.exports.all = function (test, common) {
+module.exports.all = function (test, common, listenerOpts, dialerOpts) {
```
let's add a check default to `{}` if none are passed

wanderer replied:
@diasdavid done!

diasdavid commented:
yes, like this, thank you :)

Commit:
added defaults

Commit:
added multiaddress option

wanderer commented:
@diasdavid ok I updated the PR. I also changed the location of where the option a passed to the setup method. This is because I want to use the same format for simulation where the setup method will be called multiple times.
